01-reading.Rmd

```diff
@@ -614,23 +711,148 @@ Next, we tell R what page we want to scrape by providing the webpage's URL in qu
+So we can use the `get_timelines` function to ask for the last 400 tweets from the Tidyverse account.
+```r
+rt <- get_timelines('tidyverse', n=400)
+```
+
+When you call the `get_timelines` for the first time (or any other `rtweet` function that accesses the API), 
+you will see a browser pop-up that looks something like this:
+```{r 01-tidyverse-authorize, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "The rtweet authorization prompt.", fig.retina = 2, out.width="800"}
+knitr::include_graphics("img/authorize_question.png")
+```
+
+This is the `rtweet` package asking you to provide your own Twitter account's login information.
+When `rtweet` talks to the Twitter API, it uses your account information to authenticate requests;
+Twitter then can keep track of how much data you're asking for, and how frequently you're asking.
+If you want to follow along with this example using your own Twitter account, you should read
+over the list of permissions you are granting `rtweet` *very carefully* and make sure you are comfortable
+with it. Note that `rtweet` can be used to manage most aspects of your account (make posts, add friends, etc.),
+which is why `rtweet` asks for such extensive permissions.
+If you decide to allow `rtweet` to talk to the Twitter API using your account information, then 
+input your username and password and hit "Sign In." Twitter will probably send you an email to say
+that there was an unusual login attempt on your account, and in that case you will have to take
+the one-time code they send you and provide that to the `rtweet` login page too. 
+
+> Every API has its own way to authenticate users when they try to access data. Many APIs require you to
+> sign up to receive a *token*, which is a secret password that you input into the R package (like `rtweet`) 
+> that you are using to access the API. 
+
+With the authentication setup out of the way, let's run the `get_timelines` function again to actually access
+the API and take a look at what was returned:
+```r
+rt <- get_timelines('tidyverse', n=400)
+rt
+```
+```{r 01-reading-hidden-load-tweets, echo = FALSE, message = FALSE, warning = FALSE}
+rt <- read_csv("data/tweets.csv")
+rt
+```
+The data has quite a few variables! Let's reduce this down to a few variables of interest: `created_at`, `text`, `is_retweet`, and `retweet_screen_name`.
+```{r 01-select-tweets, message = FALSE, warning = FALSE}
+rt <- rt %>% select(created_at, text, is_retweet, retweet_screen_name)
+rt
+```
+If you look back up at the image of the Tidyverse twitter page, you will
+recognize the text of the most recent few tweets in the above dataframe.  In
+other words, we have successfully created a small dataset using the Twitter
+API---neat! From this point onward, the `rt` dataframe is stored on your
+machine, and you can play with it to your heart's content; you can use
+`write_csv` to save it to a file, `read_csv` to read it into R again later, and
```

Comment:
Could we suggest they can now do more interesting things than reading and writing data? They could calculate the percentage of tweets that are retweets compared to tweets? They could find the five Twitter screen names most often retweeted by the Tidyverse Twitter account? They could create visualizations for these. These are things they do not yet know how to do, but will learn in the next two chapters. So we could state that as well (encouragement to keep reading!).

Reply:
resolved via commit